sentinel/src/lib/messages.js

```diff
@@ -2,7 +2,17 @@ var _ = require('underscore'),
     phoneNumber = require('phone-number'),
     utils = require('./utils'),
     config = require('../config');
-const messageUtils = require('@shared-libs/message-utils');
+const messageUtils = require('@shared-libs/message-utils'),
+      history = require('./history');
+
+const messageStatus = (from, msg) => {
+  let status = 'denied';
+  if(module.exports.isOutgoingAllowed(from)) {
+    const msgKey = `${msg.to}-${msg.message.replace(' ', '')}`;
```

Comment:
The `msgKey` format is specific to the history library so this is leaking the implementation. Instead `history.track` should be given two parameters `msg.to`, and `msg.message` and it should be up to this history library to concatenate them, replace whitespace, etc.

Also, why are you removing spaces?